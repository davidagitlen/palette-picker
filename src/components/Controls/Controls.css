button,
input, 
h2,
h3 {
  color: #424A60;
  font-family: 'Roboto', sans-serif;
  font-size: 18px;
  margin: 0;
}

button:hover {
  background-color: #7383C0;
  border: 1px solid #424A60;
  color: ghostwhite;
  cursor: pointer;
}

input:hover {
  border: 1px solid #424A60;
}

button {
  height: 50px;
  width: 260px;
}

input {
  display: inline-block;
  height: 48px;
  width: 255px;
  z-index: 1;
}

input::placeholder {
    padding-left: 2px;
}

h2 {
  font-family: 'Source Code Pro', monospace;
  font-size: 21px;
  margin: 10px 0 10px 20px;
  text-align: left;
}

h3 {
  font-family: 'Source Code Pro', monospace;
  font-size: 21px;
  grid-area: h;
  margin: 10px 0 5px 0px;
  text-align: left;
}

.Controls {
  display: grid;
  grid-template:
    "a . ." 1fr
    "b . ." 1fr
    "c d e" 1fr
    ". . f" .5fr
    ". . g" 1fr 
    "h . ." 1fr
    "i j k" 1fr/ 2fr .25fr 2fr;
  height: 250px;
  padding: 0 3%;
}

.random-hex-button {
  display: block;
  grid-area: a;
}

.project-name-input {
  background-image: ;
  grid-area: c;
  text-align: left;
  height: 47px;
  margin: 0 !important;
  padding: 0px 0px 0px 5px;
}

.line1 {
  grid-area: b;
  height: 35px;
  width: 260px;
  margin: auto;
}

.line2 {
  grid-area: d;
  height: 50px;
  width: 75px;
  margin: auto;
}

.line3 {
  grid-area: f;
  height: 35px;
  width: 255px;
  margin: auto;
}

.line4 {
  grid-area: j;
  height: 50px;
  width: 75px;
  margin: auto;
}

.palette-name-input {
  grid-area: e;
}

.save-button {
  grid-area: g;
}

.hex-input {
  grid-area: i;
}

.button-wrapper {
  display: flex;
  justify-content: space-between;
  grid-area: k;
}
.search-button {
  width: 48%; 
}

.clear-button {
  width: 48%; 
}

#project-name,
#project-list {
  box-shadow: none;
  font-size: 18px;
}