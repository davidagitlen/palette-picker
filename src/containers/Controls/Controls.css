button,
input, 
h2,
h3 {
  font-family: 'Roboto', sans-serif;
  margin: 0;
}

button {
  height: 50px;
  width: 260px;
}

input {
  display: inline-block;
  height: 48px;
  width: 255px;
}

h2 {
  font-size: 21px;
  margin: 10px 0 10px 20px;
  text-align: left;
}

h3 {
  grid-area: h;
  font-size: 21px;
  margin: 10px 0 5px 0px;
  text-align: left;
}

.Controls {
  display: grid;
  grid-template:
    "a . ." 1fr
    "b . ." 1fr
    "c d e" 1fr
    ". . f" .5fr
    ". . g" 1fr 
    "h . ." 1fr
    "i j k" 1fr/ 2fr .25fr 2fr;
  height: 250px;
  padding: 0 3%;
}

.random-hex-button {
  display: block;
  grid-area: a;
}

.project-name-input {
  background-image: none;
  grid-area: c;
  text-align: left;
  height: 47px;
  margin: 0 !important;
  padding: 0px 0px 0px 5px;
}

.cross1 {
  grid-area: b;
  height: 35px;
  width: 260px;
  margin: auto;
}

.cross2 {
  grid-area: d;
  height: 50px;
  width: 75px;
  margin: auto;
}

.cross3 {
  grid-area: f;
  height: 35px;
  width: 255px;
  margin: auto;
}

.cross4 {
  grid-area: j;
  height: 50px;
  width: 75px;
  margin: auto;
}

.palette-name-input {
  grid-area: e;
}

.save-button {
  grid-area: g;
}

.hex-input {
  grid-area: i;
}

.button-wrapper {
  display: flex;
  justify-content: space-between;
  grid-area: k;
}
.search-button {
  width: 48%; 
}

.clear-button {
  width: 48%; 
}