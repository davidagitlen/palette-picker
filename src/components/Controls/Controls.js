import React, { Component } from "react";
import "./Controls.css";
import cross from '../../images/cross.png';

class Controls extends Component {
  constructor(props) {
    super(props)
    this.state = {
      palette: "",
      project: "",
      hex: ""
    }
  }

  componentDidMount() {
    this.setState({projects: this.props.projects})
  }

  handleChange = e => {
    this.setState({ [e.target.name]:e.target.value })
  }

  handleRandomize = e => {
    e.preventDefault();
    this.props.getRandomHexes();
  }

  handleSave = e => {
    e.preventDefault();
    const { palette, project } = this.state;
    this.props.saveCurrentPalette(palette, project)
  }

  handleClearForm = e => {
    e.preventDefault();
    this.props.clearSearch();
    this.setState({hex : ''});
  }

  render() {
    const { projects, findPalettes } = this.props;
    const projectList = projects.map(project => {
      return <option value={project.project} key={project.id}>{project.project}</option>
    });
    return (
      <>
        <h2>Add A Project & Palette:</h2>
        <form className="Controls">
          <button className="random-hex-button" onClick={this.handleRandomize}>
            New palette!
          </button>
          <img src={cross} alt="line" className="cross1" />
          <input
            className="project-name-input"
            list="project-list"
            type="text"
            name="project"
            placeholder="Select a project!"
            autoComplete="off"
            value={this.state.project}
            onChange={this.handleChange}
          />
          <datalist id="project-list">{projectList}</datalist>
          <img src={cross} alt="line" className="cross2" />
          <input
            className="palette-name-input"
            type="text"
            name="palette"
            placeholder="Name your palette!"
            value={this.state.palette}
            onChange={this.handleChange}
          />
          <img src={cross} alt="line" className="cross3" />
          <button className="save-button" onClick={this.handleSave}>
            Save Palette
          </button>
          <h3>Search By Hex:</h3>
          <input
            className="hex-input"
            type="text"
            name="hex"
            placeholder="Enter hex code!"
            value={this.state.hex}
            onChange={this.handleChange}
          />
          <img src={cross} alt="line" className="cross4" />
          <div className="button-wrapper">
            <button
          className='search-button'
          onClick={e => findPalettes(e, this.state.hex)}
        >
          Search
        </button>
        <button
          className='clear-button'
          onClick={this.handleClearForm}
        >
          Clear
        </button>
          </div>
        </form>
      </>
    );
  }
};

export default Controls;