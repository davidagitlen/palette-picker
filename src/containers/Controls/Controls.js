import React, { Component } from "react";
import "./Controls.css";

class Controls extends Component {
  constructor(props) {
    super(props)
    this.state = {
      palette: "",
      project: ""
    }
  }

  componentDidMount() {
    this.setState({projects: this.props.projects})
  }

  handleChange = e => {
    this.setState({ [e.target.name]:e.target.value })
  }

  handleRandomize = e => {
    e.preventDefault();
    this.props.getRandomHexes();
  }

  handleSave = e => {
    e.preventDefault();
    const { palette, project } = this.state;
    this.props.saveCurrentPalette(palette, project)
  }


  render() {
    const { projects } = this.props;
    const projectList = projects.map(project => {
      return <option value={project.project} key={project.id}>{project.project}</option>
    });
    return (
      <form className="Controls">
        <h2>Controls</h2>
        <button onClick={this.handleRandomize}>New palette!</button>
        <input
          list='project-list'
          type='text'
          name='project'
          placeholder='Select a project!'
          autoComplete='off'
          value={this.state.project} 
          onChange={this.handleChange}
        />
        <datalist id='project-list'>
          {projectList}
        </datalist>
        <input
          type='text'
          name='palette'
          placeholder='Name your palette!'
          value={this.state.palette}
          onChange={this.handleChange}
        />
        <button onClick={this.handleSave}>Save Palette</button>
      </form>
    )
  }
};

export default Controls;
