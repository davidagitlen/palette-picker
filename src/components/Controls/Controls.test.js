import React from 'react';
import ReactDOM from 'react-dom';
import Controls from './Controls';

it('initial test to pass CI suite', () => {
  expect(true).toEqual(true);
});