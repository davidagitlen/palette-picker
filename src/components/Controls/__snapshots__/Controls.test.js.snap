// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match the snapshot with data passed through correctly 1`] = `
<Fragment>
  <h2>
    Add A Project & Palette:
  </h2>
  <form
    className="Controls"
  >
    <button
      className="random-hex-button"
      onClick={[Function]}
    >
      New palette!
    </button>
    <img
      alt="line"
      className="cross1"
      src="cross.png"
    />
    <input
      autoComplete="off"
      className="project-name-input"
      list="project-list"
      name="project"
      onChange={[Function]}
      placeholder="Select a project!"
      type="text"
      value=""
    />
    <datalist
      id="project-list"
    >
      <option
        key="1"
        value="Project_One"
      >
        Project_One
      </option>
      <option
        key="2"
        value="Project_Two"
      >
        Project_Two
      </option>
      <option
        key="3"
        value="Project_Three"
      >
        Project_Three
      </option>
      <option
        key="4"
        value="Project_Four"
      >
        Project_Four
      </option>
      <option
        key="5"
        value="Project_Five"
      >
        Project_Five
      </option>
    </datalist>
    <img
      alt="line"
      className="cross2"
      src="cross.png"
    />
    <input
      className="palette-name-input"
      name="palette"
      onChange={[Function]}
      placeholder="Name your palette!"
      type="text"
      value=""
    />
    <img
      alt="line"
      className="cross3"
      src="cross.png"
    />
    <button
      className="save-button"
      onClick={[Function]}
    >
      Save Palette
    </button>
    <h3>
      Search By Hex:
    </h3>
    <input
      className="hex-input"
      name="hex"
      onChange={[Function]}
      placeholder="Enter hex code!"
      type="text"
      value=""
    />
    <img
      alt="line"
      className="cross4"
      src="cross.png"
    />
    <div
      className="button-wrapper"
    >
      <button
        className="search-button"
        onClick={[Function]}
      >
        Search
      </button>
      <button
        className="clear-button"
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </form>
</Fragment>
`;
